src/index.js

```diff
@@ -0,0 +1,131 @@
+// ------------- Wave 2 ----------------------
+const state = {
+  temp: 60,
+  city: 'Seattle',
+  sky: '☁️',
+};
+// 1. increase temp
+const addDegree = (event) => {
+  state.temp += 1;
+  console.log('state temp', state.temp);
+  const tempContainer = document.querySelector('#degrees');
+  tempContainer.textContent = state.temp;
+  updateColorsAndEmojis();
+};
+// 2. decrease temp
+const subtractDegree = (event) => {
+  state.temp -= 1;
+  console.log('state temp', state.temp);
+  const tempContainer = document.querySelector('#degrees');
+  tempContainer.textContent = state.temp;
+  updateColorsAndEmojis();
+};
+// 2. temp ranges
+const updateColorsAndEmojis = () => {
+  if (state.temp > 80) {
+    document.getElementById('degrees').style.color = 'red';
+    document.getElementById('emojis').innerText = '🌵__🐍_🦂_🌵🌵__🐍_🏜_🦂';
+  } else if (state.temp >= 70 && state.temp <= 79) {
+    document.getElementById('degrees').style.color = 'orange';
+    document.getElementById('emojis').innerText = '🌸🌿🌼__🌷🌻🌿_☘️🌱_🌻🌷';
+  } else if (state.temp >= 60 && state.temp <= 69) {
+    document.getElementById('degrees').style.color = 'yellow';
+    document.getElementById('emojis').innerText = '🌾🌾_🍃_🪨__🛤_🌾🌾🌾_🍃';
+  } else if (state.temp > 50 && state.temp <= 59) {
+    document.getElementById('degrees').style.color = 'green';
+    document.getElementById('emojis').innerText = '🌲🌲';
+  } else if (state.temp < 59) {
+    document.getElementById('degrees').style.color = 'teal';
+    document.getElementById('emojis').innerText =
+      '🌲🌲⛄️🌲⛄️🍂🌲🍁🌲🌲⛄️🍂🌲';
+  }
+};
+// ------------- Wave 3 ----------------------
+const updateCity = () => {
+  let myTextInput = document.getElementById('search-bar');
+  let cityName = myTextInput.value;
+  let cityHeader = document.getElementById('city-name');
+  cityHeader.textContent = `For the lovely city of: ${cityName}`;
+  state.city = cityName;
+};
+// ------------- Wave 4 --------------------
+const findLatitudeAndLongitude = () => {
+  axios
+    .get('http://127.0.0.1:5000/location', {
+      params: {
+        q: state.city,
+        format: 'json',
+      },
+    })
+    .then((response) => {
+      console.log(response.data);
+      latitude = response.data[0].lat;
+      longitude = response.data[0].lon;
+      console.log('success in findLatitudeAndLongitude!', latitude, longitude);
+      getWeather(latitude, longitude);
+    })
+    .catch((error) => {
+      console.log('error in findLatitudeAndLongitude!', error);
+    });
```

Comment:
This works great! As an alternative, can you consider what this would look like with promise chaining? Also, in the case of a real application consider updating an element on the page to have some text telling the user there was an error and the temperature couldn't be updated.